internal/host/static/repository_host.go

```diff
@@ -23,44 +23,45 @@ import (
 // unique within h.CatalogId.
 func (r *Repository) CreateHost(ctx context.Context, scopeId string, h *Host, opt ...Option) (*Host, error) {
 	if h == nil {
-		return nil, fmt.Errorf("create: static host: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithMsg("nil host"))
```

Comment:
So the open question here is do we want to move away at this point from this generic `InvalidParameter` and return a specific `NilHost` error with this PR.  This would be a step to having a clear error code/ID for each possible error returned.

Simialarly below we would then have `NoCatalogId`, `NoScopeId` etc

Reply:
I think returning `InvalidParameter` is fine for all of the checks except for the `ErrInvalidAddress` check.